File: src/1175.prime-arrangements.1279/solution.ts
function numPrimeArrangements (n: number): number {
    const p = countPrimesLT(n + 1);
    const mod = 10 ** 9 + 7;
    let result = 1;
    for (let i = 2; i <= p; i++) {
        result = result * i % mod;
    }
    for (let i = 2; i <= n - p; i++) {
        result = result * i % mod;
    }
    return result;
}

function countPrimesLT (n:number) {
    const isPrimes = new Array<boolean>(n).fill(true);
    let result = 0;
    for (let i = 2; i < n; i++) {
        if (!isPrimes[i]) {
            continue;
        }
        result++;
        for (let j = 2 * i; j < n; j += i) {
            isPrimes[j] = false;
        }
    }
    return result;
}

File: src/1175.prime-arrangements.1279/solution.java
class Solution {
    private final int mod = 1000000007;
    public int numPrimeArrangements(int n) {
        int[] primes = new int[]{2, 3, 5, 7, 11, 13, 17, 19, 23, 29, 31, 37, 41, 43, 47, 53, 59, 61, 67, 71, 73, 79, 83, 89, 97};
        int prime = 0;
        for(int i=0;i<primes.length;i++){
            if(primes[i]>n){
                break;
            }
            prime++;
        }

        return (int)(factorial(prime)*factorial(n-prime)%mod);
    }

    private long factorial(int n){
        long result = 1;
        while (n>0){
            result = (result*n)%mod;
            n--;
        }
        return result;
    }

}